shell/run_shell_cmd.go

```diff
@@ -33,30 +34,47 @@ func RunShellCommand(terragruntOptions *options.TerragruntOptions, command strin
 
 	cmd.Dir = terragruntOptions.WorkingDir
 
-	signalChannel := NewSignalsForwarder(forwardSignals, cmd.Process, terragruntOptions.Logger)
+	cmdChannel := make(chan error)
+	signalChannel := NewSignalsForwarder(forwardSignals, cmd, terragruntOptions.Logger, cmdChannel)
 	defer signalChannel.Close()
 
-	return errors.WithStackTrace(cmd.Run())
+	err := cmd.Run()
+	cmdChannel <- err
+
+	return errors.WithStackTrace(err)
+}
+
+
+// Return the exit code of a command. If the error is not an exec.ExitError type,
+// the error is returned.
+func GetExitCode(err error) (int, error) {
+	var retCode int
+	if exiterr, ok := errors.Unwrap(err).(*exec.ExitError); ok {
+	status := exiterr.Sys().(syscall.WaitStatus)
+	retCode = status.ExitStatus()
+	return retCode, nil
```

Comment:
In fact, I wouldn't declare `retCode` at all. Just return `status.ExitStatus(), nil` inside the if-statement and return `0` outside of it.